crates/rpc-server/src/registry.rs

```diff
@@ -581,35 +603,48 @@ async fn submit_l2transaction(
 
     let l2tx_bytes = l2tx.into_bytes();
     let tx = packed::L2Transaction::from_slice(&l2tx_bytes)?;
-    let tx_hash = to_jsonh256(tx.hash().into());
+
+    // fetch mem-pool state
+    let db = store.begin_transaction();
+    let tree = db.mem_pool_state_tree()?;
+
     // check sender's nonce
-    {
-        // fetch mem-pool state
-        let db = store.begin_transaction();
-        let tree = db.mem_pool_state_tree()?;
-        // sender_id
-        let sender_id = tx.raw().from_id().unpack();
-        let sender_nonce: u32 = tree.get_nonce(sender_id)?;
-        let tx_nonce: u32 = tx.raw().nonce().unpack();
-        if sender_nonce != tx_nonce {
-            let err = TransactionError::Nonce {
-                account_id: sender_id,
-                expected: sender_nonce,
-                actual: tx_nonce,
-            };
-            log::info!(
-                "[RPC] reject to submit tx {:?}, err: {}",
-                faster_hex::hex_string(&tx.hash()),
-                err
-            );
-            return Err(RpcError::Full {
-                code: INVALID_NONCE_ERR_CODE,
+    let sender_id = tx.raw().from_id().unpack();
+    let sender_nonce: u32 = tree.get_nonce(sender_id)?;
+    let tx_nonce: u32 = tx.raw().nonce().unpack();
+    if sender_nonce != tx_nonce {
+        let err = TransactionError::Nonce {
+            account_id: sender_id,
+            expected: sender_nonce,
+            actual: tx_nonce,
+        };
+        log::info!(
+            "[RPC] reject to submit tx {:?}, err: {}",
+            faster_hex::hex_string(&tx.hash()),
+            err
+        );
+        return Err(RpcError::Full {
+            code: INVALID_NONCE_ERR_CODE,
+            message: err.to_string(),
+            data: None,
+        });
+    }
+
+    // check tx fee or gasPrice of the l2tx
+    let raw_l2tx = tx.raw();
+    let backend_type = get_backend_type(tree, generator, &raw_l2tx)?;
+    gw_utils::fee::check_l2tx_fee(&mem_pool_config.fee_config, &raw_l2tx, backend_type).map_err(
+        |err| {
+            log::warn!("check_fee_ret err: {}", err);
```

Comment:
```suggestion
            log::debug!("check_fee_ret err: {}", err);
```

This is a normal behaviour, not a warning for Godwoken program